### DISCUSSION TITLE
Add minNumber, minSpecial and ambiguous password generation options to CLI

### DISCUSSION THREAD

**Discussion by jomifepe** [bitwarden/clients#5975] [Password Manager]
### Code Contribution Proposal

Hello, I noticed that new options for password generation have been added to multiple clients, but the CLI is still missing those. Options like `Minimum numbers`, `Minimum special`, and `Avoid ambiguous characters`.

I created a PR to add them: https://github.com/bitwarden/clients/pull/5974

Code changes are pretty minor because these options already exist, just aren't available on the CLI,

**Comment by dwbit** [comment]
Thanks for the proposal @jomifepe, the team will review and get back to you shortly!

**Comment by jomifepe** [comment]
Hello, the PR has been open for a while with no response, and I noticed that some of the more recent community PRs are making progress. I was just curious if the CLI is considered a client with slightly lower priority 🙂

**Reply by djsmith85** [reply]
We have seen your PR and all contributions go through a review process by product/design before passed on to engineering. The tools team will be responsible for code review. Glancing at it though without any kind of testing. Does the `convertNumberOption` guard against wrong input (i.e. alphanumeric) and provide an error message?

**Reply by jomifepe** [reply]
Thank you for answering and taking a look! 

The `convertNumberOption` is essentially doing exactly what's [currently being done](https://github.com/bitwarden/clients/blob/5c576fd19e7c8fad208db84e490420725b124474/apps/cli/src/tools/generate.command.ts#L58) inline in the `generate.command.ts`, it was just a way of reducing complexity and make the CodeScene tool happy. Although, I just added validation for alphanumeric input, thanks for the suggestion. It does not provide an error message when the input is invalid, it fallbacks to the default value, same for all the other parameters.

**Comment by djsmith85** [comment]
Closing this discussion, as this was included with `2023.9`
Thanks again @jomifepe  for your contributions ❤️
